hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/ZookeeperOperations.java

```diff
@@ -0,0 +1,69 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import pl.allegro.tech.hermes.common.exception.InternalProcessingException;
+
+import java.util.List;
+import java.util.Optional;
+
+class ZookeeperOperations {
+
+    private final CuratorFramework curator;
+
+    ZookeeperOperations(CuratorFramework curator) {
+        this.curator = curator;
+    }
+
+    void writeOrCreatePersistent(String path, byte[] serializedData) throws Exception {
+        try {
+            curator.setData().forPath(path, serializedData);
+        } catch (KeeperException.NoNodeException e) {
+            try {
+                curator.create().creatingParentContainersIfNeeded()
+                        .withMode(CreateMode.PERSISTENT)
```

Comment:
You can use `orSetData()` method here

Reply:
I'm afraid it's not available in the curator version used. :(